File: php/src/Controllers/AdvertisementController.php
<?php
declare(strict_types=1);

namespace Demo\App\Controllers;

use Demo\App\Framework\Database\SqliteConnection;
use Demo\App\Framework\FrameworkRequest;
use Demo\App\Framework\FrameworkResponse;
use Demo\App\Model\AdvertisementModel;

final readonly class AdvertisementController
{
    public function __construct(private SqliteConnection $connection)
    {
    }

    public function addAdvertisement(FrameworkRequest $request): FrameworkResponse
    {
        $advertisement = new AdvertisementModel(
            ($request->content())['id'],
            ($request->content())['description'],
            ($request->content())['password'],
        );

        $dbConnection = $this->connection->connect();

        $dbConnection->exec(sprintf("INSERT INTO advertisements (id, description, password) VALUES ('%s', '%s', '%s');",
                $advertisement->id(),
                $advertisement->description(),
                md5($advertisement->password()),
            )
        );

        return new FrameworkResponse(
            FrameworkResponse::STATUS_CREATED,
            []
        );
    }
}

File: kotlin/src/main/kotlin/controllers/AdvertisementController.kt
package controllers

import framework.FrameworkRequest
import framework.FrameworkResponse
import framework.database.DatabaseConnection
import model.AdvertisementModel
import java.security.MessageDigest

class AdvertisementController (private val connection: DatabaseConnection) {
    fun addAdvertisement(request: FrameworkRequest): FrameworkResponse {
        val advertisement = AdvertisementModel(
            request.content["id"],
            request.content["description"],
            request.content["password"],
            )

        val passwordHash = advertisement.password!!.md5()
        connection.execute(
            "INSERT INTO advertisements (id, description, password) VALUES ('${advertisement.id}', '${advertisement.description}', '$passwordHash')"
        )

        return FrameworkResponse(
            FrameworkResponse::STATUS_CREATED,
            mapOf(),
        )
    }

    private fun String.md5(): String {
        val md = MessageDigest.getInstance("MD5")
        val digest = md.digest(this.toByteArray())
        return digest.toString()
    }
}

